api/types/types.proto

```diff
@@ -1543,3 +1542,40 @@ message DeleteWebTokenRequest {
     // Token specifies the token to delete.
     string Token = 2 [ (gogoproto.jsontag) = "token" ];
 }
+
+// EnvironmentVariable describes an environment variable with a key and value.
+message EnvironmentVariable {
+    // The name of the variable.
+    string Key = 1 [ (gogoproto.jsontag) = "key" ];
+
+    // The value of the variable.
+    string Value = 2 [ (gogoproto.jsontag) = "value" ];
+}
+
+// PAMConfigV3 represents a cluster-wide PAM configuration.
+message PAMConfigV3 {
+    // Kind is a resource kind.
+    string Kind = 1 [ (gogoproto.jsontag) = "kind" ];
+    // SubKind is an optional resource sub kind, used in some resources.
+    string SubKind = 2 [ (gogoproto.jsontag) = "sub_kind,omitempty" ];
+    // Version is version.
+    string Version = 3 [ (gogoproto.jsontag) = "version" ];
+    // Metadata is a resource metadata.
+    Metadata Metadata = 4 [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "metadata" ];
+    // Spec is a PAM config specification.
+    PAMConfigSpecV3 Spec = 5 [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "spec" ];
+}
+
+// PAMConfigSpecV3 specifies the PAM configuration.
+// Fields in this configuration are optional and will default to local configuration if not
+// specified.
+message PAMConfigSpecV3 {
+    // Whether PAM integration is enabled. Defaults to false.
+    bool Enabled = 1 [ (gogoproto.jsontag) = "enabled" ];
+    // The PAM service name, defaults to defaults.PAMServiceName.
+    string ServiceName = 2 [ (gogoproto.jsontag) = "service_name" ];
+    // Whether PAM authentication is enabled. Defaults to false.
+    bool UsePAMAuth = 3 [ (gogoproto.jsontag) = "use_pam_auth" ];
+    // Environment mappings set for PAM modules.
+    repeated EnvironmentVariable Environment = 4 [ (gogoproto.jsontag) = "environment,omitempty" ];
```

Comment:
Any reason this is a array instead of a map? It's being used as a map internally from what I can tell.

Reply:
Last I checked the protobuf spec, it did not have a native map type and this is the recommended solution. Do we do anything special that gives us a map type?

Reply:
Ah, it looks like gogoproto supports it, was not aware of it.

Reply:
Addressed